### PR TITLE
kubeadm: increase robustness for kubeadm etcd operations

### LINKED ISSUE #2181: make the etcd client retry logic consistent
after this merged:
https://github.com/kubernetes/kubernetes/pull/90645

we made the MemberAdd retry logic a bit different in terms of retries.

- make etcd client logic consistent for all methods, by wrapping both client dial and client operation
- use the same timeouts / exp-backoff everywhere.

/priority important-longterm
/kind cleanup

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind bug
/kind cleanup

**What this PR does / why we need it**:
make the etcd client retry logic consistent, increase robustness for kubeadm etcd operations.

**Which issue(s) this PR fixes**:

Fixes kubernetes/kubeadm#2181

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/util/etcd/etcd.go [hunk 1]
```diff
@@ -217,22 +217,25 @@ const dialTimeout = 40 * time.Second
...
-
 	// Syncs the list of endpoints
+	var cli *clientv3.Client
 	var lastError error
-	err = wait.ExponentialBackoff(etcdBackoff, func() (bool, error) {
+	err := wait.ExponentialBackoff(etcdBackoff, func() (bool, error) {
+		var err error
+		cli, err = clientv3.New(clientv3.Config{
+			Endpoints:   c.Endpoints,
+			DialTimeout: dialTimeout,
```
Comment (neolit123) [inline cmd/kubeadm/app/util/etcd/etcd.go:227]: for MemberAdd we made this equal to `etcdTimeout`, which means that we can potentially remove the `dialTimeout` var.

the note on https://github.com/kubernetes/kubernetes/blob/64c70a7f0f4a55398b6bc7fc1be0fde3dcaa97bb/cmd/kubeadm/app/util/etcd/etcd.go#L213-L216

is concerning, but maybe our new retries scheme would be beneficial for the arm64 issue too.

**Review by neolit123 — COMMENTED** [review]
thanks @SataQiu .
added one comment to make the dial timeout consistent too.

/assign @fabriziopandini

File: cmd/kubeadm/app/util/etcd/etcd.go (hunk 1 above)
Comment (SataQiu) [reply cmd/kubeadm/app/util/etcd/etcd.go]: +1  let's do it.
If there's a problem, we can revert it back.

**Comment by neolit123** [comment]
/lgtm
defering to @fabriziopandini for approval.

**Pull request merged**